### PR TITLE
Make it CRUDy

### PR DESCRIPTION
**Pull request by thorrester**
### Short Summary
Adds ability to delete cards

### Context
Currently, `opsml` allows users to `create`, `read` and `update cards`. This PR adds the method `delete_card` that allows a user to supply a `card` to delete.

### Is this a Breaking Change?
No. adds a new method

### CODE REVIEW

File: opsml/app/routes/cards.py [hunk 1]
```diff
@@ -155,3 +157,23 @@ def update_card(
     logger.info("Updated card: %s", str(payload.model_dump()))
 
     return UpdateCardResponse(updated=True)
+
+
+@router.post(
+    "/cards/delete",
+    response_model=DeleteCardResponse,
+    name="update_card",
```
Comment (damonallison) [inline opsml/app/routes/cards.py:165]: delete_card?

File: opsml/app/routes/cards.py [hunk 2]
```diff
@@ -155,3 +157,23 @@ def update_card(
...
+    payload: DeleteCardRequest = Body(...),
+) -> DeleteCardResponse:
+    """Deletes a specific artifact card"""
+    table_for_registry = payload.table_name.split("_")[1].lower()
+    registry: CardRegistry = getattr(request.app.state.registries, table_for_registry)
+    registry._registry.delete_card_record(card=payload.card)
+
+    logger.info("Deleted card: %s", str(payload.model_dump()))
+
+    return DeleteCardResponse(deleted=True)
```
Comment (damonallison) [inline opsml/app/routes/cards.py:179]: Is this ever false? Seems like when it's false errors are raised. Perhaps we check for existence before attempting to delete / return false if not found?

File: opsml/registry/sql/base/client.py [hunk 3]
```diff
@@ -141,6 +141,20 @@ def update_card_record(self, card: Dict[str, Any]) -> Tuple[Dict[str, Any], str]
             return card, "updated"
         raise ValueError("Failed to update card")
 
+    @log_card_change
```
Comment (damonallison) [inline opsml/registry/sql/base/client.py:144]: 💯

**Review by damonallison — APPROVED** [review]

File: opsml/app/routes/cards.py (hunk 2 above)
Comment (thorrester) [reply opsml/app/routes/cards.py:179]: This is for the record in postgres/mysql. The actual artifact deleting is done on another path, but you're right. The current storage clients all do a soft fail if nothing is found, but I think we'll want to inform if no files are found.

**Pull request merged**